moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp

```diff
@@ -263,7 +204,74 @@ void PlanningSceneMonitor::initialize(const planning_scene::PlanningScenePtr& sc
   state_update_timer_ =
       node_->create_wall_timer(dt_state_update_, std::bind(&PlanningSceneMonitor::stateUpdateTimerCallback, this));
 
-  // reconfigure_impl_ = new DynamicReconfigureImpl(this);
+  // Set up publishing parameters
+  rcl_interfaces::msg::ParameterDescriptor desc;
+  desc.set__type(rclcpp::ParameterType::PARAMETER_BOOL);
+
+  desc.set__name("publish_planning_scene");
+  desc.set__description("Set to True to publish Planning Scenes");
+  bool publish_planning_scene = node_->declare_parameter("planning_scene_monitor.publish_planning_scene", false, desc);
+  
+  desc.set__name("publish_geometry_updates");
+  desc.set__description("Set to True to publish geometry updates of the planning scene");
+  bool publish_geom_updates = node_->declare_parameter("planning_scene_monitor.publish_geometry_updates", true, desc);
+
+  desc.set__name("publish_state_updates");
+  desc.set__description("Set to True to publish state updates of the planning scene");
+  bool publish_state_updates = node_->declare_parameter("planning_scene_monitor.publish_state_updates", false, desc);
+
+  desc.set__name("publish_transforms_updates");
+  desc.set__description("Set to True to publish transform updates of the planning scene");
+  bool publish_transform_updates = node_->declare_parameter("planning_scene_monitor.publish_transforms_updates", false, desc);
+
+  desc.set__type(rclcpp::ParameterType::PARAMETER_DOUBLE);
+  desc.set__name("publish_planning_scene_hz");
+  desc.set__description("Set the maximum frequency at which planning scene updates are published");
+  double publish_planning_scene_hz = node_->declare_parameter("planning_scene_monitor.publish_planning_scene_hz", 4.0, desc);
```

Comment:
We have to check if the parameter is declared for the node before declaring it otherwise it will throw an exception, for example, if we create two monitors for the same node

Reply:
has_parameter check before declare/get parameter 7f381b0